### PR TITLE
Indexing SPL Governance accounts

### PR DESCRIPTION
**Pull request by imabdulbasit**

### CODE REVIEW

File: crates/indexer/src/geyser/accounts/spl_governance.rs [hunk 1]
```diff
@@ -0,0 +1,713 @@
...
+    for o in data.options {
+        let row = DbProposalOption {
+            proposal_address: Owned(key.to_string()),
+            label: Owned(o.label.to_string()),
+            vote_weight: o.vote_weight.try_into()?,
+            vote_result: match o.vote_result {
+                OptionVoteResult::None => OptionVoteResultEnum::None,
+                OptionVoteResult::Succeeded => OptionVoteResultEnum::Succeeded,
+                OptionVoteResult::Defeated => OptionVoteResultEnum::Defeated,
+            },
```
Comment (kespinola) [inline crates/indexer/src/geyser/accounts/spl_governance.rs:663]: NB: Can these matches be bundled into a from trait?

File: crates/indexer/src/geyser/accounts/spl_governance.rs [hunk 2]
```diff
@@ -0,0 +1,713 @@
...
+    write_version: u64,
+) -> Result<()> {
+    let (vote_threshold_type, vote_threshold_percentage) =
+        if let Some(VoteThresholdPercentage::YesVote(p)) = data.vote_threshold_percentage {
+            (Some(VoteThresholdEnum::YesVote), Some(i16::try_from(p)?))
+        } else if let Some(VoteThresholdPercentage::Quorum(p)) = data.vote_threshold_percentage {
+            (Some(VoteThresholdEnum::Quorum), Some(i16::try_from(p)?))
+        } else {
+            (None, None)
+        };
```
Comment (kespinola) [inline crates/indexer/src/geyser/accounts/spl_governance.rs:580]: This seems like a good item for the Relms team to review, yeah?

**Review by kespinola — COMMENTED** [review]
looking good.

File: crates/indexer/src/geyser/accounts/spl_governance.rs (hunk 2 above)
Comment (imabdulbasit) [reply crates/indexer/src/geyser/accounts/spl_governance.rs:580]: > This seems like a good item for the Relms team to review, yeah?

Its an Option<VoteThresholdPercentage> which I have broken into two columns of a table.
https://docs.rs/spl-governance/2.1.1/spl_governance/state/enums/enum.VoteThresholdPercentage.html

**Review by mpwsh — APPROVED** [review]

**Pull request merged**